### DISCUSSION TITLE
Concurrent vs Parallel?

### DISCUSSION THREAD

**Discussion by alash3al** [spatie/fork#14] [Q&A]
This package uses `pcntl_fork` which uses `fork(2)` to create a child process based on php.net docs.

A child process means it may be executed in parallel not concurrently, Am I wrong?

**Comment by brendt** [comment]
> concurrent execution - the execution of two or more computer programs by a single computer

> parallel programming - This type of programming takes a problem, breaks it down into a series of smaller steps, delivers instructions, and processors execute the solutions at the same time

I'd say both definitions match? What's your interpretation of "concurrent" and "parallel" ?
